### DISCUSSION TITLE
Tokio version 1.38.1 introducing excessive memory allocations, is this normal?

### DISCUSSION THREAD

**Discussion by viciousstar** [tokio-rs/tokio#6756] [General] [answered]
In version 1.38, our memory allocation QPS was at 1k/s, but after upgrading to 1.38.1, it became 3k/s. Currently, we are not sure if this has any impact on the overall performance of our application.

**Comment by Darksonn** [comment, marked as answer]
The 1.38.1 release contained a critical bugfix, but it was discovered during review that it introduced an additional allocation: [link](https://github.com/tokio-rs/tokio/pull/6683#discussion_r1677175494). If this is making a big difference, then that is very useful feedback. Please open an issue referring to that discussion requesting that the allocations are eliminated.
